src/System.Configuration/src/System/Configuration/ImplicitMachineConfigHost.cs

```diff
@@ -66,7 +66,22 @@ public override Stream OpenStreamForRead(string streamName)
 @"<configuration>
     <configSections>
         <section name='appSettings' type='System.Configuration.AppSettingsSection, System.Configuration' restartOnExternalChanges='false' requirePermission='false'/>
+        <section name='connectionStrings' type='System.Configuration.ConnectionStringsSection, System.Configuration' requirePermission='false'/>
+        <section name='mscorlib' type='System.Configuration.IgnoreSection, System.Configuration' allowLocation='false'/>
+        <section name='runtime' type='System.Configuration.IgnoreSection, System.Configuration' allowLocation='false'/>
+        <section name='assemblyBinding' type='System.Configuration.IgnoreSection, System.Configuration' allowLocation='false'/>
+        <section name='satelliteassemblies' type='System.Configuration.IgnoreSection, System.Configuration' allowLocation='false'/>
+        <section name='startup' type='System.Configuration.IgnoreSection, System.Configuration' allowLocation='false'/>
     </configSections>
+    <configProtectedData defaultProvider='RsaProtectedConfigurationProvider'>
+        <providers>
+            <add name = 'RsaProtectedConfigurationProvider' type='System.Configuration.RsaProtectedConfigurationProvider,System.Configuration' description='Uses RsaCryptoServiceProvider to encrypt and decrypt' keyContainerName='NetFrameworkConfigurationKey' cspProviderName='' useMachineContainer='true' useOAEP='false'/>
+            <add name = 'DataProtectionConfigurationProvider' type='System.Configuration.DpapiProtectedConfigurationProvider,System.Configuration' description='Uses CryptProtectData and CryptUnProtectData Windows APIs to encrypt and decrypt' useMachineProtection='true' keyEntropy=''/>
+        </providers>
+    </configProtectedData>
+    <connectionStrings>
+        <add name = 'LocalSqlServer' connectionString='data source=.\SQLEXPRESS;Integrated Security=SSPI;AttachDBFilename=|DataDirectory|aspnetdb.mdf;User Instance=true' providerName='System.Data.SqlClient'/>
```

Comment:
We have it on desktop- while some code may not port without it, my biggest issue is that we end up with null properties off of Config if we don't have something here.

My rule-of-thumb is try to make built in types work seamlessly. That includes being able to instantiate all System.Config types and avoid exposing null properties.